spotipy/client.py

```diff
@@ -1588,10 +1579,21 @@ def _search_multiple_markets(self, q, limit, offset, type, markets, total):
                 UserWarning,
             )
 
+        results = {}
+        first_type = type.split(",")[0] + 's'
+        count = 0
+
         for country in markets:
             result = self._get(
                 "search", q=q, limit=limit, offset=offset, type=type, market=country
             )
             results[country] = result
 
+            count += len(result[first_type]['items'])
+            if total and count >= total:
+                return results
```

Comment:
Would we need to slice `result[first_type]['items']`? Like I was doing before? In the case that `count += len(result[first_type]['items'])` sends it over the total.

Reply:
No, `limit` should never be higher than `total` thanks to auto-adjusting `limit` at the top and bottom of this method

Reply:
Right - that is correct